specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/searchindex.json

```diff
@@ -224,10 +543,88 @@
           }
         }
       }
-      
     }
   },
   "definitions": {
+    "HitHighlights": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        }
+      },
+      "description": "Defines the generic document in an Azure Search index.",
+      "x-ms-external": true
+    },
+    "FacetResults": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        }
+      },
+      "x-ms-external": true
+    },
+    "DocumentSearchResult": {
+      "type": "object",
+        "properties": {
+          "value": {
+            "type": "array",
+            "items": {
+                "$ref": "#/definitions/SearchDocument"
+            },
+            "x-ms-client-name": "Results"
+          },
+          "@odata.count": {
+            "type": "integer",
+            "format": "int64",
+            "x-ms-client-name": "Count"
+          },
+          "@search.coverage": {
+            "type": "number",
+            "format": "double",
+            "x-ms-client-name": "Coverage"
+          },
+          "@search.facets": {
+            "type": "object",
+            "$ref": "#/definitions/FacetResults",
+            "x-ms-client-name": "Facets"
+          },
+          "@odata.nextLink": {
+            "type": "string",
+            "x-ms-client-name": "NextLink"
+          },
+          "@search.nextPageParameters": {
+            "$ref": "#/definitions/SearchParametersPayload",
+            "x-ms-client-name": "NextSearchParametersPayload"
+          }
+        }
+    },
+    "SearchDocument": {
+      "type": "object",
+      "properties": {
+         "@search.score": {
+            "type": "number",
+            "format": "double",
+            "x-ms-client-name": "Score",
+            "x-nullable": false
+         },
+         "@search.highlights": {
+            "$ref": "#/definitions/HitHighlights",
+            "x-ms-client-name": "Highlights"
+         }
+       },
+       "additionalProperties": true,
+       "x-ms-external": true
+    },
+    "IndexBatch": {
+      "properties": {
+        "name": {
+          "type": "string"
+        }
+      },
+      "x-ms-external": true
```

Comment:
This still needs to be fully modeled (IndexAction, action enum)

Reply:
Yes- Will do.

Reply:
Yes- I believe more can be generated. I will address it in a subsequent commit, in the same PR. 

---
In reply to: [208105464](https://github.com/Azure/azure-rest-api-specs/pull/3578#discussion_r208105464) [](ancestors = 208105464)